### PR TITLE
Welder ext: handle deletes and errors

### PR DESCRIPTION
**Pull request by calbach**
- [x] I've documented my API changes in Swagger

In all cases:

- [x] Get a thumbsworth of review and PO signoff if necessary
- [x] Verify all tests go green
- [ ] Squash and merge; you can delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: docker/jupyter/custom/jupyter_delocalize.py [hunk 1]
```diff
@@ -178,21 +178,30 @@ def __init__(self, *args, **kwargs):
...
         pass
-      raise IOError("safeDelocalize failed: " + msg)
+
+      if resp.status_code == 412 and error_json.get('errorCode', -1) == 1:
+        # 1: Storage Link not found; expected for unmanaged files.
+        return
+
+      msg = resp.reason or 'unknown Welder error'
+      if error_json != {}:
+        msg = json.dumps(error_json)
```
Comment (Qi77Qi) [inline docker/jupyter/custom/jupyter_delocalize.py]: just curious, do error messages get logged somewhere?

Comment (calbach) [reply docker/jupyter/custom/jupyter_delocalize.py]: I see them getting logged by Tornado / Jupyter locally, but worth confirming once we're able to deploy this fully.

File: docker/jupyter/custom/jupyter_delocalize.py [hunk 2]
```diff
@@ -231,5 +239,5 @@ def rename_file(self, old_path, new_path):
     super(WelderContentsManager, self).rename_file(old_path, new_path)
 
   def delete_file(self, path):
-    # TODO(IA-1049): Integrate Welder delete.
+    self._post_welder('delete', path)
```
Comment (Qi77Qi) [inline docker/jupyter/custom/jupyter_delocalize.py]: if welder delete fails, will we still delete the local file?

Comment (calbach) [reply docker/jupyter/custom/jupyter_delocalize.py]: No - this line with throw an exception. I have a test case to cover this situation.

**Pull request merged**